### ISSUE TITLE
Error Message on start and crash on exit

### ISSUE THREAD

**Issue by e-orz** [zealdocs/zeal#595]
Hi,
I am using the latest build (commit a4fac24a5dd9cb3df12794cd5485fb5daedf23f6 from 15/9/2016).
I am on Ubuntu 16.04.
Zeal crashes on exit with the error below. The issue wasn't with the previous build dated several months before.
It seams that it causes also an error on start:
"Another application instance can be still running, or has crashed.
Make sure to start Zeal only once."
pressing Retry helps but the error is for each start of the application.

The error on exit:
```
*** Error in `./zeal': free(): invalid next size (normal): 0x0000000002360c80 ***
======= Backtrace: =========
/lib/x86_64-linux-gnu/libc.so.6(+0x77725)[0x7fc706823725]
/lib/x86_64-linux-gnu/libc.so.6(+0x7ff4a)[0x7fc70682bf4a]
/lib/x86_64-linux-gnu/libc.so.6(cfree+0x4c)[0x7fc70682fabc]
./zeal[0x421809]
./zeal[0x421839]
./zeal[0x41f001]
/lib/x86_64-linux-gnu/libc.so.6(__libc_start_main+0xf0)[0x7fc7067cc830]
./zeal[0x41f5c9]
======= Memory map: ========
00400000-004af000 r-xp 00000000 08:05 1074607567                         /home/eli/dev/zeal/bin/zeal
006ae000-006b1000 r--p 000ae000 08:05 1074607567                         /home/eli/dev/zeal/bin/zeal
006b1000-006b3000 rw-p 000b1000 08:05 1074607567                         /home/eli/dev/zeal/bin/zeal
01e6e000-024dc000 rw-p 00000000 00:00 0                                  [heap]
7fc668000000-7fc668021000 rw-p 00000000 00:00 0 
7fc668021000-7fc66c000000 ---p 00000000 00:00 0 
7fc670000000-7fc670021000 rw-p 00000000 00:00 0 
7fc670021000-7fc674000000 ---p 00000000 00:00 0 
7fc674000000-7fc674021000 rw-p 00000000 00:00 0 
7fc674021000-7fc678000000 ---p 00000000 00:00 0 
7fc678000000-7fc678021000 rw-p 00000000 00:00 0 
7fc678021000-7fc67c000000 ---p 00000000 00:00 0 
7fc67c000000-7fc67c021000 rw-p 00000000 00:00 0 
7fc67c021000-7fc680000000 ---p 00000000 00:00 0 
7fc680000000-7fc680657000 rw-p 00000000 00:00 0 
7fc680657000-7fc684000000 ---p 00000000 00:00 0 
7fc684000000-7fc684021000 rw-p 00000000 00:00 0 
7fc684021000-7fc688000000 ---p 00000000 00:00 0 
7fc688000000-7fc688021000 rw-p 00000000 00:00 0 
7fc688021000-7fc68c000000 ---p 00000000 00:00 0 
7fc68e8aa000-7fc68e8b0000 ---p 00000000 00:00 0 
7fc68e8b0000-7fc68e8c0000 rw-p 00000000 00:00 0 
7fc68e8c0000-7fc68e8d0000 ---p 00000000 00:00 0 
7fc68e8d0000-7fc68e8e0000 rw-p 00000000 00:00 0 
7fc68e8e0000-7fc68e8f0000 ---p 00000000 00:00 0 
7fc68e8f0000-7fc68e900000 rw-p 00000000 00:00 0 
7fc68e900000-7fc68e910000 ---p 00000000 00:00 0 
7fc68e910000-7fc68e920000 rw-p 00000000 00:00 0 
7fc68e920000-7fc68e930000 ---p 00000000 00:00 0 
7fc68e930000-7fc68e940000 rw-p 00000000 00:00 0 
7fc68e940000-7fc68e950000 ---p 00000000 00:00 0 
7fc68e950000-7fc68e960000 rw-p 00000000 00:00 0 
7fc68e960000-7fc68e970000 ---p 00000000 00:00 0 
7fc68e970000-7fc68e980000 rw-p 00000000 00:00 0 
7fc68e980000-7fc68e990000 ---p 00000000 00:00 0 
7fc68e990000-7fc68e9a0000 rw-p 00000000 00:00 0 
7fc68e9a0000-7fc68e9b0000 ---p 00000000 00:00 0 
7fc68e9b0000-7fc68e9c0000 rw-p 00000000 00:00 0 
7fc68e9c0000-7fc68e9d0000 ---p 00000000 00:00 0 
7fc68e9d0000-7fc68e9f0000 rw-p 00000000 00:00 0 
7fc68e9f0000-7fc68ea00000 ---p 00000000 00:00 0 
7fc68ea00000-7fc68ea10000 rw-p 00000000 00:00 0 
7fc68ea10000-7fc68ea20000 ---p 00000000 00:00 0 
7fc68ea20000-7fc68ea30000 rw-p 00000000 00:00 0 
7fc68ea30000-7fc68ea40000 ---p 00000000 00:00 0 
7fc68ea40000-7fc68ea50000 rw-p 00000000 00:00 0 
7fc68ea50000-7fc68ea5c000 ---p 00000000 00:00 0 
7fc68efb5000-7fc68efba000 r-xp 00000000 00:15 814465                     /lib/x86_64-linux-gnu/libnss_dns-2.23.so
7fc68efba000-7fc68f1ba000 ---p 00005000 00:15 814465                     /lib/x86_64-linux-gnu/libnss_dns-2.23.so
7fc68f1ba000-7fc68f1bb000 r--p 00005000 00:15 814465                     /lib/x86_64-linux-gnu/libnss_dns-2.23.so
7fc68f1bb000-7fc68f1bc000 rw-p 00006000 00:15 814465                     /lib/x86_64-linux-gnu/libnss_dns-2.23.so
7fc68f1bc000-7fc68f1be000 r-xp 00000000 00:15 1034605                    /lib/x86_64-linux-gnu/libnss_mdns4_minimal.so.2
7fc68f1be000-7fc68f3bd000 ---p 00002000 00:15 1034605                    /lib/x86_64-linux-gnu/libnss_mdns4_minimal.so.2
7fc68f3bd000-7fc68f3be000 r--p 00001000 00:15 1034605                    /lib/x86_64-linux-gnu/libnss_mdns4_minimal.so.2
7fc68f3be000-7fc68f3bf000 rw-p 00002000 00:15 1034605                    /lib/x86_64-linux-gnu/libnss_mdns4_minimal.so.2
7fc68f3bf000-7fc68f3c0000 ---p 00000000 00:00 0 
7fc68f3c0000-7fc68fbc0000 rw-p 00000000 00:00 0 
7fc68fbc0000-7fc68fbcb000 r-xp 00000000 00:15 814453                     /lib/x86_64-linux-gnu/libnss_files-2.23.so
7fc68fbcb000-7fc68fdca000 ---p 0000b000 00:15 814453                     /lib/x86_64-linux-gnu/libnss_files-2.23.so
7fc68fdca000-7fc68fdcb000 r--p 0000a000 00:15 814453                     /lib/x86_64-linux-gnu/libnss_files-2.23.so
7fc68fdcb000-7fc68fdcc000 rw-p 0000b000 00:15 814453                     /lib/x86_64-linux-gnu/libnss_files-2.23.so
7fc68fdcc000-7fc68fdd2000 rw-p 00000000 00:00 0 
7fc68fddc000-7fc68fde0000 ---p 00000000 00:00 0 
7fc68fde0000-7fc68fdf0000 rw-p 00000000 00:00 0 
7fc68fdf0000-7fc68fdfb000 ---p 00000000 00:00 0 
7fc68fdfb000-7fc68fdfd000 r-xp 00000000 00:15 852107                     /usr/lib/x86_64-linux-gnu/libproxy/0.4.11/modules/network_networkmanager.so
7fc68fdfd000-7fc68fffc000 ---p 00002000 00:15 852107                     /usr/lib/x86_64-linux-gnu/libproxy/0.4.11/modules/network_networkmanager.so
7fc68fffc000-7fc68fffd000 r--p 00001000 00:15 852107                     /usr/lib/x86_64-linux-gnu/libproxy/0.4.11/modules/network_networkmanager.so
7fc68fffd000-7fc68fffe000 rw-p 00002000 00:15 852107                     /usr/lib/x86_64-linux-gnu/libproxy/0.4.11/modules/network_networkmanager.so
7fc68fffe000-7fc68ffff000 ---p 00000000 00:00 0 
7fc68ffff000-7fc690000000 rwxp 00000000 00:00 0 
7fc690000000-7fc6d0000000 ---p 00000000 00:00 0 
7fc6d0000000-7fc6d0022000 rw-p 00000000 00:00 0 
7fc6d0022000-7fc6d4000000 ---p 00000000 00:00 0 
7fc6d4007000-7fc6d4010000 ---p 00000000 00:00 0 
7fc6d4010000-7fc6d4020000 rw-p 00000000 00:00 0 
7fc6d4020000-7fc6d4026000 ---p 00000000 00:00 0 
7fc6d4026000-7fc6d4030000 r-xp 00000000 00:15 852122                     /usr/lib/x86_64-linux-gnu/libproxy/0.4.11/modules/config_gnome3.so
7fc6d4030000-7fc6d422f000 ---p 0000a000 00:15 852122                     /usr/lib/x86_64-linux-gnu/libproxy/0.4.11/modules/config_gnome3.so
7fc6d422f000-7fc6d4230000 r--p 00009000 00:15 852122                     /usr/lib/x86_64-linux-gnu/libproxy/0.4.11/modules/config_gnome3.so
7fc6d4230000-7fc6d4231000 rw-p 0000a000 00:15 852122                     /usr/lib/x86_64-linux-gnu/libproxy/0.4.11/modules/config_gnome3.so
7fc6d4231000-7fc6d4232000 ---p 00000000 00:00 0 
7fc6d4232000-7fc6d4a32000 rw-p 00000000 00:00 0 
7fc6d4a32000-7fc6d4aeb000 r--p 00000000 00:15 55541                      /usr/share/fonts/truetype/dejavu/DejaVuSans.ttf
7fc6d4aeb000-7fc6d4b42000 r--p 00000000 00:15 55544                      /usr/share/fonts/truetype/dejavu/DejaVuSerif-Bold.ttf
7fc6d4b42000-7fc6d4b9f000 r--p 00000000 00:15 55545                      /usr/share/fonts/truetype/dejavu/DejaVuSerif.ttf
7fc6d4b9f000-7fc6d4dce000 r-xp 00000000 00:15 1704050                    /lib/x86_64-linux-gnu/libcrypto.so.1.0.0
7fc6d4dce000-7fc6d4fcd000 ---p 0022f000 00:15 1704050                    /lib/x86_64-linux-gnu/libcrypto.so.1.0.0
7fc6d4fcd000-7fc6d4fe9000 r--p 0022e000 00:15 1704050                    /lib/x86_64-linux-gnu/libcrypto.so.1.0.0
7fc6d4fe9000-7fc6d4ff6000 rw-p 0024a000 00:15 1704050                    /lib/x86_64-linux-gnu/libcrypto.so.1.0.0
7fc6d4ff6000-7fc6d4ffa000 rw-p 00000000 00:00 0 
7fc6d4ffa000-7fc6d4ffb000 ---p 00000000 00:00 0 
7fc6d4ffb000-7fc6d57fb000 rw-p 00000000 00:00 0 
7fc6d57fb000-7fc6d57fc000 ---p 00000000 00:00 0 
7fc6d57fc000-7fc6d5ffc000 rw-p 00000000 00:00 0 
7fc6d5ffc000-7fc6d5ffd000 ---p 00000000 00:00 0 
7fc6d5ffd000-7fc6d67fd000 rw-p 00000000 00:00 0 
7fc6d67fd000-7fc6d67fe000 ---p 00000000 00:00 0 
7fc6d67fe000-7fc6d6ffe000 rw-p 00000000 00:00 0 
7fc6d6ffe000-7fc6d6fff000 ---p 00000000 00:00 0 
7fc6d6fff000-7fc6d77ff000 rw-p 00000000 00:00 0 
7fc6d77ff000-7fc6d7800000 ---p 00000000 00:00 0 
7fc6d7800000-7fc6d8000000 rw-p 00000000 00:00 0 
7fc6d8000000-7fc6d8031000 rw-p 00000000 00:00 0 
7fc6d8031000-7fc6dc000000 ---p 00000000 00:00 0 
7fc6dc012000-7fc6dc071000 r-xp 00000000 00:15 1704051                    /lib/x86_64-linux-gnu/libssl.so.1.0.0
7fc6dc071000-7fc6dc270000 ---p 0005f000 00:15 1704051                    /lib/x86_64-linux-gnu/libssl.so.1.0.0
7fc6dc270000-7fc6dc274000 r--p 0005e000 00:15 1704051                    /lib/x86_64-linux-gnu/libssl.so.1.0.0
7fc6dc274000-7fc6dc27b000 rw-p 00062000 00:15 1704051                    /lib/x86_64-linux-gnu/libssl.so.1.0.0
7fc6dc281000-7fc6dc2a4000 r--s 00000000 00:15 1775264                    /usr/share/mime/mime.cache
7fc6dc2a4000-7fc6dc2b0000 ---p 00000000 00:00 0 
7fc6dc2b0000-7fc6dc2c0000 rw-p 00000000 00:00 0 
7fc6dc2c0000-7fc6dc2d0000 ---p 00000000 00:00 0 
7fc6dc2d0000-7fc6dc2e0000 rw-p 00000000 00:00 0 
7fc6dc2e0000-7fc6dc2e2000 ---p 00000000 00:00 0 
7fc6dc2e2000-7fc6dc2e6000 rw-p 00000000 00:00 0 
7fc6dc2e6000-7fc6dc6f0000 ---p 00000000 00:00 0 
7fc6dc6f0000-7fc6dc700000 rw-p 00000000 00:00 0 
7fc6dc700000-7fc6dc710000 ---p 00000000 00:00 0 
7fc6dc710000-7fc6dc720000 rw-p 00000000 00:00 0 
7fc6dc720000-7fc6dc729000 r-xp 00000000 00:15 904896                     /usr/lib/x86_64-linux-gnu/qt5/plugins/iconengines/libqsvgicon.so
7fc6dc729000-7fc6dc72a000 r--p 00008000 00:15 904896                     /usr/lib/x86_64-linux-gnu/qt5/plugins/iconengines/libqsvgicon.so
7fc6dc72a000-7fc6dc72b000 rw-p 00009000 00:15 904896                     /usr/lib/x86_64-linux-gnu/qt5/plugins/iconengines/libqsvgicon.so
7fc6dc72b000-7fc6dc733000 rw-s 00000000 08:05 1613098628                 /home/eli/.local/share/Zeal/Zeal/docsets/Java_SE8.docset/Contents/Resources/docSet.dsidx-shm
7fc6dc733000-7fc6dc79e000 rw-p 00000000 00:00 0 
7fc6dc79e000-7fc6dc7ab000 r-xp 00000000 00:15 1255000                    /usr/lib/x86_64-linux-gnu/qt5/plugins/sqldrivers/libqsqlite.so
7fc6dc7ab000-7fc6dc7ac000 ---p 0000d000 00:15 1255000                    /usr/lib/x86_64-linux-gnu/qt5/plugins/sqldrivers/libqsqlite.so
7fc6dc7ac000-7fc6dc7ad000 r--p 0000d000 00:15 1255000                    /usr/lib/x86_64-linux-gnu/qt5/plugins/sqldrivers/libqsqlite.so
7fc6dc7ad000-7fc6dc7ae000 rw-p 0000e000 00:15 1255000                    /usr/lib/x86_64-linux-gnu/qt5/plugins/sqldrivers/libqsqlite.so
7fc6dc7ae000-7fc6dc7ec000 r-xp 00000000 00:15 1254230                    /usr/lib/x86_64-linux-gnu/qt5/plugins/bearer/libqnmbearer.so
7fc6dc7ec000-7fc6dc7ee000 r--p 0003d000 00:15 1254230                    /usr/lib/x86_64-linux-gnu/qt5/plugins/bearer/libqnmbearer.so
7fc6dc7ee000-7fc6dc7ef000 rw-p 0003f000 00:15 1254230                    /usr/lib/x86_64-linux-gnu/qt5/plugins/bearer/libqnmbearer.so
7fc6dc7ef000-7fc6dc7f0000 ---p 00000000 00:00 0 
7fc6dc7f0000-7fc6dcff0000 rw-p 00000000 00:00 0 
7fc6dcff0000-7fc6dcff1000 ---p 00000000 00:00 0 
7fc6dcff1000-7fc6dd0f1000 rw-p 00000000 00:00 0 
7fc6dd0f1000-7fc6dd0f2000 ---p 00000000 00:00 0 
7fc6dd0f2000-7fc6dd152000 rw-s 00000000 00:05 335970388                  /SYSV00000000 (deleted)
7fc6dd152000-7fc6dd186000 r-xp 00000000 00:15 1254228                    /usr/lib/x86_64-linux-gnu/qt5/plugins/bearer/libqconnmanbearer.so
7fc6dd186000-7fc6dd187000 ---p 00034000 00:15 1254228                    /usr/lib/x86_64-linux-gnu/qt5/plugins/bearer/libqconnmanbearer.so
7fc6dd187000-7fc6dd188000 r--p 00034000 00:15 1254228                    /usr/lib/x86_64-linux-gnu/qt5/plugins/bearer/libqconnmanbearer.so
7fc6dd188000-7fc6dd189000 rw-p 00035000 00:15 1254228                    /usr/lib/x86_64-linux-gnu/qt5/plugins/bearer/libqconnmanbearer.so
7fc6dd189000-7fc6dd191000 r-xp 00000000 00:15 30459                      /usr/lib/x86_64-linux-gnu/libpciaccess.so.0.11.1
7fc6dd191000-7fc6dd391000 ---p 00008000 00:15 30459                      /usr/lib/x86_64-linux-gnu/libpciaccess.so.0.11.1
7fc6dd391000-7fc6dd392000 r--p 00008000 00:15 30459                      /usr/lib/x86_64-linux-gnu/libpciaccess.so.0.11.1
7fc6dd392000-7fc6dd393000 rw-p 00009000 00:15 30459                      /usr/lib/x86_64-linux-gnu/libpciaccess.so.0.11.1
7fc6dd393000-7fc6dd39e000 r-xp 00000000 00:15 1704267                    /usr/lib/x86_64-linux-gnu/libdrm_radeon.so.1.0.1
7fc6dd39e000-7fc6dd59d000 ---p 0000b000 00:15 1704267                    /usr/lib/x86_64-linux-gnu/libdrm_radeon.so.1.0.1
7fc6dd59d000-7fc6dd59e000 r--p 0000a000 00:15 1704267                    /usr/lib/x86_64-linux-gnu/libdrm_radeon.so.1.0.1
7fc6dd59e000-7fc6dd59f000 rw-p 0000b000 00:15 1704267                    /usr/lib/x86_64-linux-gnu/libdrm_radeon.so.1.0.1
7fc6dd59f000-7fc6dd5a6000 r-xp 00000000 00:15 1704303                    /usr/lib/x86_64-linux-gnu/libdrm_nouveau.so.2.0.0
7fc6dd5a6000-7fc6dd7a5000 ---p 00007000 00:15 1704303                    /usr/lib/x86_64-linux-gnu/libdrm_nouveau.so.2.0.0
7fc6dd7a5000-7fc6dd7a6000 r--p 00006000 00:15 1704303                    /usr/lib/x86_64-linux-gnu/libdrm_nouveau.so.2.0.0
7fc6dd7a6000-7fc6dd7a7000 rw-p 00007000 00:15 1704303                    /usr/lib/x86_64-linux-gnu/libdrm_nouveau.so.2.0.0
7fc6dd7a7000-7fc6dd7c7000 r-xp 00000000 00:15 1704231                    /usr/lib/x86_64-linux-gnu/libdrm_intel.so.1.0.0
7fc6dd7c7000-7fc6dd9c6000 ---p 00020000 00:15 1704231                    /usr/lib/x86_64-linux-gnu/libdrm_intel.so.1.0.0
7fc6dd9c6000-7fc6dd9c7000 r--p 0001f000 00:15 1704231                    /usr/lib/x86_64-linux-gnu/libdrm_intel.so.1.0.0
7fc6dd9c7000-7fc6dd9c8000 rw-p 00020000 00:15 1704231                    /usr/lib/x86_64-linux-gnu/libdrm_intel.so.1.0.0
7fc6dd9c8000-7fc6ddfcd000 r-xp 00000000 00:15 1778104                    /usr/lib/x86_64-linux-gnu/dri/i965_dri.so
7fc6ddfcd000-7fc6de1cd000 ---p 00605000 00:15 1778104                    /usr/lib/x86_64-linux-gnu/dri/i965_dri.so
7fc6de1cd000-7fc6de203000 r--p 00605000 00:15 1778104                    /usr/lib/x86_64-linux-gnu/dri/i965_dri.so
7fc6de203000-7fc6de20a000 rw-p 0063b000 00:15 1778104                    /usr/lib/x86_64-linux-gnu/dri/i965_dri.so
7fc6de20a000-7fc6de212000 rw-p 00000000 00:00 0 
7fc6de212000-7fc6de230000 r-xp 00000000 00:15 1606801                    /lib/x86_64-linux-gnu/libudev.so.1.6.4
7fc6de230000-7fc6de231000 r--p 0001d000 00:15 1606801                    /lib/x86_64-linux-gnu/libudev.so.1.6.4
7fc6de231000-7fc6de232000 rw-p 0001e000 00:15 1606801                    /lib/x86_64-linux-gnu/libudev.so.1.6.4
7fc6de232000-7fc6de289000 r--p 00000000 00:15 906496                     /usr/share/fonts/truetype/ubuntu-font-family/Ubuntu-R.ttf
7fc6de289000-7fc6de2c0000 r-xp 00000000 00:15 856293                     /usr/lib/x86_64-linux-gnu/libcroco-0.6.so.3.0.1
7fc6de2c0000-7fc6de4c0000 ---p 00037000 00:15 856293                     /usr/lib/x86_64-linux-gnu/libcroco-0.6.so.3.0.1
7fc6de4c0000-7fc6de4c3000 r--p 00037000 00:15 856293                     /usr/lib/x86_64-linux-gnu/libcroco-0.6.so.3.0.1
7fc6de4c3000-7fc6de4c4000 rw-p 0003a000 00:15 856293                     /usr/lib/x86_64-linux-gnu/libcroco-0.6.so.3.0.1
7fc6de4c4000-7fc6de4f9000 r-xp 00000000 00:15 856563                     /usr/lib/x86_64-linux-gnu/librsvg-2.so.2.40.13
7fc6de4f9000-7fc6de6f9000 ---p 00035000 00:15 856563                     /usr/lib/x86_64-linux-gnu/librsvg-2.so.2.40.13
7fc6de6f9000-7fc6de6fa000 r--p 00035000 00:15 856563                     /usr/lib/x86_64-linux-gnu/librsvg-2.so.2.40.13
7fc6de6fa000-7fc6de6fb000 rw-p 00036000 00:15 856563                     /usr/lib/x86_64-linux-gnu/librsvg-2.so.2.40.13
7fc6de700000-7fc6de70d000 r-xp 00000000 00:15 1254229                    /usr/lib/x86_64-linux-gnu/qt5/plugins/bearer/libqgenericbearer.so
7fc6de70d000-7fc6de70e000 ---p 0000d000 00:15 1254229                    /usr/lib/x86_64-linux-gnu/qt5/plugins/bearer/libqgenericbearer.so
7fc6de70e000-7fc6de70f000 r--p 0000d000 00:15 1254229                    /usr/lib/x86_64-linux-gnu/qt5/plugins/bearer/libqgenericbearer.so
7fc6de70f000-7fc6de710000 rw-p 0000e000 00:15 1254229                    /usr/lib/x86_64-linux-gnu/qt5/plugins/bearer/libqgenericbearer.so
7fc6de710000-7fc6de711000 ---p 00000000 00:00 0 
7fc6de711000-7fc6de719000 rw-p 00000000 00:00 0 
7fc6de719000-7fc6de71b000 ---p 00000000 00:00 0 
7fc6de71b000-7fc6de723000 rw-p 00000000 00:00 0 
7fc6de723000-7fc6de724000 ---p 00000000 00:00 0 
7fc6de724000-7fc6de726000 r-xp 00000000 00:15 856538                     /usr/lib/x86_64-linux-gnu/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-svg.so
7fc6de726000-7fc6de925000 ---p 00002000 00:15 856538                     /usr/lib/x86_64-linux-gnu/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-svg.so
7fc6de925000-7fc6de926000 r--p 00001000 00:15 856538                     /usr/lib/x86_64-linux-gnu/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-svg.so
7fc6de926000-7fc6de927000 rw-p 00002000 00:15 856538                     /usr/lib/x86_64-linux-gnu/gdk-pixbuf-2.0/2.10.0/loaders/libpixbufloader-svg.so
7fc6de927000-7fc6de928000 ---p 00000000 00:00 0 
7fc6de928000-7fc6df128000 rw-p 00000000 00:00 0 
7fc6df128000-7fc6df129000 ---p 00000000 00:00 0 
7fc6df129000-7fc6df929000 rw-p 00000000 00:00 0 
7fc6df929000-7fc6df95f000 r-xp 00000000 00:15 1291029                    /usr/lib/x86_64-linux-gnu/gvfs/libgvfscommon.so
7fc6df95f000-7fc6dfb5f000 ---p 00036000 00:15 1291029                    /usr/lib/x86_64-linux-gnu/gvfs/libgvfscommon.so
7fc6dfb5f000-7fc6dfb64000 r--p 00036000 00:15 1291029                    /usr/lib/x86_64-linux-gnu/gvfs/libgvfscommon.so
7fc6dfb64000-7fc6dfb65000 rw-p 0003b000 00:15 1291029                    /usr/lib/x86_64-linux-gnu/gvfs/libgvfscommon.so
7fc6dfb65000-7fc6dfb95000 r-xp 00000000 00:15 1290900                    /usr/lib/x86_64-linux-gnu/gio/modules/libgvfsdbus.so
7fc6dfb95000-7fc6dfd94000 ---p 00030000 00:15 1290900                    /usr/lib/x86_64-linux-gnu/gio/modules/libgvfsdbus.so
7fc6dfd94000-7fc6dfd95000 r--p 0002f000 00:15 1290900                    /usr/lib/x86_64-linux-gnu/gio/modules/libgvfsdbus.so
7fc6dfd95000-7fc6dfd96000 rw-p 00030000 00:15 1290900                    /usr/lib/x86_64-linux-gnu/gio/modules/libgvfsdbus.so
7fc6dfd96000-7fc6dfd97000 rw-p 00000000 00:00 0 
7fc6dfd97000-7fc6dfdfd000 r-xp 00000000 00:15 878986                     /usr/lib/x86_64-linux-gnu/libibus-1.0.so.5.0.511
7fc6dfdfd000-7fc6dfffc000 ---p 00066000 00:15 878986                     /usr/lib/x86_64-linux-gnu/libibus-1.0.so.5.0.511
7fc6dfffc000-7fc6dfffe000 r--p 00065000 00:15 878986                     /usr/lib/x86_64-linux-gnu/libibus-1.0.so.5.0.511
7fc6dfffe000-7fc6dffff000 rw-p 00067000 00:15 878986                     /usr/lib/x86_64-linux-gnu/libibus-1.0.so.5.0.511
7fc6dffff000-7fc6e0000000 rw-p 00000000 00:00 0 
7fc6e0000000-7fc6e0021000 rw-p 00000000 00:00 0 
7fc6e0021000-7fc6e4000000 ---p 00000000 00:00 0 
7fc6e4000000-7fc6e4022000 rw-p 00000000 00:00 0 
7fc6e4022000-7fc6e8000000 ---p 00000000 00:00 0 
7fc6e8000000-7fc6e8022000 rw-p 00000000 00:00 0 
7fc6e8022000-7fc6ec000000 ---p 00000000 00:00 0 
7fc6ec002000-7fc6ec003000 ---p 00000000 00:00 0 
7fc6ec003000-7fc6ec00b000 rw-p 00000000 00:00 0 
7fc6ec00b000-7fc6ec00d000 ---p 00000000 00:00 0 
7fc6ec00d000-7fc6ec015000 rw-p 00000000 00:00 0 
7fc6ec015000-7fc6ec016000 ---p 00000000 00:00 0 
7fc6ec016000-7fc6ec039000 r--s 00000000 00:15 1775264                    /usr/share/mime/mime.cache
7fc6ec039000-7fc6ec03f000 r-xp 00000000 00:15 1027087                    /usr/lib/x86_64-linux-gnu/gtk-2.0/2.10.0/immodules/im-ibus.so
7fc6ec03f000-7fc6ec23f000 ---p 00006000 00:15 1027087                    /usr/lib/x86_64-linux-gnu/gtk-2.0/2.10.0/immodules/im-ibus.so
7fc6ec23f000-7fc6ec240000 r--p 00006000 00:15 1027087                    /usr/lib/x86_64-linux-gnu/gtk-2.0/2.10.0/immodules/im-ibus.so
7fc6ec240000-7fc6ec241000 rw-p 00007000 00:15 1027087                    /usr/lib/x86_64-linux-gnu/gtk-2.0/2.10.0/immodules/im-ibus.so
7fc6ec241000-7fc6ec2c0000 r-xp 00000000 00:15 925357                     /usr/lib/x86_64-linux-gnu/libgmp.so.10.3.0
7fc6ec2c0000-7fc6ec4bf000 ---p 0007f000 00:15 925357                     /usr/lib/x86_64-linux-gnu/libgmp.so.10.3.0
7fc6ec4bf000-7fc6ec4c0000 r--p 0007e000 00:15 925357                     /usr/lib/x86_64-linux-gnu/libgmp.so.10.3.0
7fc6ec4c0000-7fc6ec4c1000 rw-p 0007f000 00:15 925357                     /usr/lib/x86_64-linux-gnu/libgmp.so.10.3.0
7fc6ec4c1000-7fc6ec4f3000 r-xp 00000000 00:15 839648                     /usr/lib/x86_64-linux-gnu/libhogweed.so.4.2
7fc6ec4f3000-7fc6ec6f2000 ---p 00032000 00:15 839648                     /usr/lib/x86_64-linux-gnu/libhogweed.so.4.2
7fc6ec6f2000-7fc6ec6f3000 r--p 00031000 00:15 839648                     /usr/lib/x86_64-linux-gnu/libhogweed.so.4.2Aborted (core dumped)
```

**Comment by trollixx** [comment]
Could you build Zeal in the debug mode to get a meaningful backtrace? Just run `qmake CONFIG+=debug` instead of just `qmake`.

The warning is about a local socket file left from the crashed instance. When you click `Retry`, Zeal removes the old one.

**Comment by e-orz** [comment]
I compiled it again with debug, but the output is pretty much the same.
I may be missing some dependencies?

here is the compile output:
```
cd assets/ && ( test -e Makefile || /usr/lib/x86_64-linux-gnu/qt5/bin/qmake /home/eli/dev/zeal/assets/assets.pro CONFIG+=debug -o Makefile ) && make -f Makefile 
Project MESSAGE: Browser engine: Qt WebKit.
Project MESSAGE: Portable build: No.
Project MESSAGE: Install prefix: /usr
make[1]: Entering directory '/home/eli/dev/zeal/assets'
make[1]: Nothing to be done for 'first'.
make[1]: Leaving directory '/home/eli/dev/zeal/assets'
cd src/ && ( test -e Makefile || /usr/lib/x86_64-linux-gnu/qt5/bin/qmake /home/eli/dev/zeal/src/src.pro CONFIG+=debug -o Makefile ) && make -f Makefile 
make[1]: Entering directory '/home/eli/dev/zeal/src'
cd libs/ && ( test -e Makefile || /usr/lib/x86_64-linux-gnu/qt5/bin/qmake /home/eli/dev/zeal/src/libs/libs.pro CONFIG+=debug -o Makefile ) && make -f Makefile 
make[2]: Entering directory '/home/eli/dev/zeal/src/libs'
cd core/ && ( test -e Makefile || /usr/lib/x86_64-linux-gnu/qt5/bin/qmake /home/eli/dev/zeal/src/libs/core/core.pro CONFIG+=debug -o Makefile ) && make -f Makefile 
Project MESSAGE: Browser engine: Qt WebKit.
Project MESSAGE: Portable build: No.
Project MESSAGE: Install prefix: /usr
make[3]: Entering directory '/home/eli/dev/zeal/src/libs/core'
compiling application.cpp
compiling extractor.cpp
compiling localserver.cpp
compiling settings.cpp
moc application.h
compiling ../../../.moc/moc_application.cpp
moc extractor.h
compiling ../../../.moc/moc_extractor.cpp
moc localserver.h
compiling ../../../.moc/moc_localserver.cpp
moc settings.h
compiling ../../../.moc/moc_settings.cpp
rm -f libCore.a
ar cqs libCore.a ../../../.obj/application.o ../../../.obj/extractor.o ../../../.obj/localserver.o ../../../.obj/settings.o ../../../.obj/moc_application.o ../../../.obj/moc_extractor.o ../../../.obj/moc_localserver.o ../../../.obj/moc_settings.o
rm -f ../../../.lib/libCore.a
mv -f libCore.a ../../../.lib/ 
make[3]: Leaving directory '/home/eli/dev/zeal/src/libs/core'
cd registry/ && ( test -e Makefile || /usr/lib/x86_64-linux-gnu/qt5/bin/qmake /home/eli/dev/zeal/src/libs/registry/registry.pro CONFIG+=debug -o Makefile ) && make -f Makefile 
Project MESSAGE: Browser engine: Qt WebKit.
Project MESSAGE: Portable build: No.
Project MESSAGE: Install prefix: /usr
make[3]: Entering directory '/home/eli/dev/zeal/src/libs/registry'
compiling cancellationtoken.cpp
compiling docset.cpp
compiling docsetmetadata.cpp
compiling docsetregistry.cpp
compiling listmodel.cpp
compiling searchmodel.cpp
compiling searchquery.cpp
compiling searchresult.cpp
moc docsetregistry.h
compiling ../../../.moc/moc_docsetregistry.cpp
moc listmodel.h
compiling ../../../.moc/moc_listmodel.cpp
moc searchmodel.h
compiling ../../../.moc/moc_searchmodel.cpp
rm -f libRegistry.a
ar cqs libRegistry.a ../../../.obj/cancellationtoken.o ../../../.obj/docset.o ../../../.obj/docsetmetadata.o ../../../.obj/docsetregistry.o ../../../.obj/listmodel.o ../../../.obj/searchmodel.o ../../../.obj/searchquery.o ../../../.obj/searchresult.o ../../../.obj/moc_docsetregistry.o ../../../.obj/moc_listmodel.o ../../../.obj/moc_searchmodel.o
rm -f ../../../.lib/libRegistry.a
mv -f libRegistry.a ../../../.lib/ 
make[3]: Leaving directory '/home/eli/dev/zeal/src/libs/registry'
cd ui/ && ( test -e Makefile || /usr/lib/x86_64-linux-gnu/qt5/bin/qmake /home/eli/dev/zeal/src/libs/ui/ui.pro CONFIG+=debug -o Makefile ) && make -f Makefile 
Project MESSAGE: Browser engine: Qt WebKit.
Project MESSAGE: Portable build: No.
Project MESSAGE: Install prefix: /usr
Project MESSAGE: AppIndicator support: Yes.
make[3]: Entering directory '/home/eli/dev/zeal/src/libs/ui'
uic forms/aboutdialog.ui
uic forms/docsetsdialog.ui
uic forms/mainwindow.ui
uic forms/settingsdialog.ui
compiling aboutdialog.cpp
compiling docsetlistitemdelegate.cpp
compiling docsetsdialog.cpp
compiling mainwindow.cpp
compiling progressitemdelegate.cpp
compiling searchitemdelegate.cpp
compiling settingsdialog.cpp
compiling widgets/searchablewebview.cpp
compiling widgets/searchedit.cpp
compiling widgets/shortcutedit.cpp
compiling widgets/toolbarframe.cpp
compiling widgets/webview.cpp
compiling qxtglobalshortcut/qxtglobalshortcut.cpp
compiling qxtglobalshortcut/qxtglobalshortcut_x11.cpp
moc aboutdialog.h
compiling ../../../.moc/moc_aboutdialog.cpp
moc docsetlistitemdelegate.h
compiling ../../../.moc/moc_docsetlistitemdelegate.cpp
moc docsetsdialog.h
compiling ../../../.moc/moc_docsetsdialog.cpp
moc mainwindow.h
compiling ../../../.moc/moc_mainwindow.cpp
moc progressitemdelegate.h
compiling ../../../.moc/moc_progressitemdelegate.cpp
moc searchitemdelegate.h
compiling ../../../.moc/moc_searchitemdelegate.cpp
moc settingsdialog.h
compiling ../../../.moc/moc_settingsdialog.cpp
moc widgets/searchablewebview.h
compiling ../../../.moc/moc_searchablewebview.cpp
moc widgets/searchedit.h
compiling ../../../.moc/moc_searchedit.cpp
moc widgets/toolbarframe.h
compiling ../../../.moc/moc_toolbarframe.cpp
moc widgets/webview.h
compiling ../../../.moc/moc_webview.cpp
moc qxtglobalshortcut/qxtglobalshortcut.h
compiling ../../../.moc/moc_qxtglobalshortcut.cpp
rm -f libUi.a
ar cqs libUi.a ../../../.obj/aboutdialog.o ../../../.obj/docsetlistitemdelegate.o ../../../.obj/docsetsdialog.o ../../../.obj/mainwindow.o ../../../.obj/progressitemdelegate.o ../../../.obj/searchitemdelegate.o ../../../.obj/settingsdialog.o ../../../.obj/searchablewebview.o ../../../.obj/searchedit.o ../../../.obj/shortcutedit.o ../../../.obj/toolbarframe.o ../../../.obj/webview.o ../../../.obj/qxtglobalshortcut.o ../../../.obj/qxtglobalshortcut_x11.o ../../../.obj/moc_aboutdialog.o ../../../.obj/moc_docsetlistitemdelegate.o ../../../.obj/moc_docsetsdialog.o ../../../.obj/moc_mainwindow.o ../../../.obj/moc_progressitemdelegate.o ../../../.obj/moc_searchitemdelegate.o ../../../.obj/moc_settingsdialog.o ../../../.obj/moc_searchablewebview.o ../../../.obj/moc_searchedit.o ../../../.obj/moc_toolbarframe.o ../../../.obj/moc_webview.o ../../../.obj/moc_qxtglobalshortcut.o
rm -f ../../../.lib/libUi.a
mv -f libUi.a ../../../.lib/ 
make[3]: Leaving directory '/home/eli/dev/zeal/src/libs/ui'
cd util/ && ( test -e Makefile || /usr/lib/x86_64-linux-gnu/qt5/bin/qmake /home/eli/dev/zeal/src/libs/util/util.pro CONFIG+=debug -o Makefile ) && make -f Makefile 
Project MESSAGE: Browser engine: Qt WebKit.
Project MESSAGE: Portable build: No.
Project MESSAGE: Install prefix: /usr
make[3]: Entering directory '/home/eli/dev/zeal/src/libs/util'
compiling plist.cpp
compiling version.cpp
rm -f libUtil.a
ar cqs libUtil.a ../../../.obj/plist.o ../../../.obj/version.o
rm -f ../../../.lib/libUtil.a
mv -f libUtil.a ../../../.lib/ 
make[3]: Leaving directory '/home/eli/dev/zeal/src/libs/util'
make[2]: Leaving directory '/home/eli/dev/zeal/src/libs'
cd app/ && ( test -e Makefile || /usr/lib/x86_64-linux-gnu/qt5/bin/qmake /home/eli/dev/zeal/src/app/app.pro CONFIG+=debug -o Makefile ) && make -f Makefile 
Project MESSAGE: Browser engine: Qt WebKit.
Project MESSAGE: Portable build: No.
Project MESSAGE: Install prefix: /usr
Cannot read /home/eli/dev/zeal/src/libs/Makefile/Makefile.pri: Not a directory
Project MESSAGE: AppIndicator support: Yes.
make[2]: Entering directory '/home/eli/dev/zeal/src/app'
compiling main.cpp
rcc resources/zeal.qrc
compiling ../../.rcc/qrc_zeal.cpp
linking ../../bin/zeal
make[2]: Leaving directory '/home/eli/dev/zeal/src/app'
make[1]: Leaving directory '/home/eli/dev/zeal/src'
```

**Comment by trollixx** [comment]
Thanks for trying. The backtrace should have had references to the actual
lines of code. If it's the same (addresses only), then the only way would
be to run Zeal in a debugger. Could you do that? Dealing with gdb directly
is not the most pleasant experience, but it's very easy to do via Qt
Creator.

**Comment by e-orz** [comment]
I pulled to commit bdaba34a8c0bf70513040d9d9505f42c37825ede

I went to the GDB way. If it won't be enough I will install QT Creator.
I will try to add more details:
I am using regularly zeal compiled few months ago. Everything is ok.
When first (and only first) time running the new version after the old one the process crashes with segmentation fault. Stacktace using GDB (with line numbers):
```
Thread 1 "zeal" received signal SIGSEGV, Segmentation fault.
0x00007ffff34fd348 in QHashData::rehash(int) () from /usr/lib/x86_64-linux-gnu/libQt5Core.so.5
(gdb) where
#0  0x00007ffff34fd348 in QHashData::rehash(int) () from /usr/lib/x86_64-linux-gnu/libQt5Core.so.5
#1  0x00007ffff643d9a4 in QWidgetPrivate::init(QWidget*, QFlags<Qt::WindowType>) () from /usr/lib/x86_64-linux-gnu/libQt5Widgets.so.5
#2  0x00007ffff64f92cb in QAbstractSlider::QAbstractSlider(QAbstractSliderPrivate&, QWidget*) () from /usr/lib/x86_64-linux-gnu/libQt5Widgets.so.5
#3  0x00007ffff658b5d8 in QScrollBar::QScrollBar(Qt::Orientation, QWidget*) () from /usr/lib/x86_64-linux-gnu/libQt5Widgets.so.5
#4  0x00007ffff65bcd74 in ?? () from /usr/lib/x86_64-linux-gnu/libQt5Widgets.so.5
#5  0x00007ffff65be4dc in QAbstractScrollAreaPrivate::init() () from /usr/lib/x86_64-linux-gnu/libQt5Widgets.so.5
#6  0x00007ffff66528b9 in QAbstractItemView::QAbstractItemView(QAbstractItemViewPrivate&, QWidget*) () from /usr/lib/x86_64-linux-gnu/libQt5Widgets.so.5
#7  0x00007ffff6692266 in QTreeView::QTreeView(QWidget*) () from /usr/lib/x86_64-linux-gnu/libQt5Widgets.so.5
#8  0x000000000043cbd3 in Ui_MainWindow::setupUi (this=0x96ac80, MainWindow=0x95d300) at ../../../.ui/ui_mainwindow.h:163
#9  0x000000000042f466 in MainWindow::MainWindow (this=0x95d300, app=0x932400, parent=0x0) at mainwindow.cpp:185
#10 0x000000000042319e in Zeal::Core::Application::Application (this=0x932400, parent=0x0) at application.cpp:79
#11 0x0000000000420938 in main (argc=1, argv=0x7fffffffde58) at main.cpp:264
```
On later invocations this error disappears but for each exit there is segmenation fault. GDB stacktrace:
```
Thread 1 "zeal" received signal SIGABRT, Aborted.
0x00007ffff2ac8418 in __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:54
54	../sysdeps/unix/sysv/linux/raise.c: No such file or directory.
(gdb) where
#0  0x00007ffff2ac8418 in __GI_raise (sig=sig@entry=6) at ../sysdeps/unix/sysv/linux/raise.c:54
#1  0x00007ffff2aca01a in __GI_abort () at abort.c:89
#2  0x00007ffff2b0a72a in __libc_message (do_abort=do_abort@entry=2, fmt=fmt@entry=0x7ffff2c236b0 "*** Error in `%s': %s: 0x%s ***\n") at ../sysdeps/posix/libc_fatal.c:175
#3  0x00007ffff2b12f4a in malloc_printerr (ar_ptr=<optimized out>, ptr=<optimized out>, str=0x7ffff2c23808 "free(): invalid next size (normal)", action=3) at malloc.c:5007
#4  _int_free (av=<optimized out>, p=<optimized out>, have_lock=0) at malloc.c:3868
#5  0x00007ffff2b16abc in __GI___libc_free (mem=<optimized out>) at malloc.c:2969
#6  0x0000000000431162 in MainWindow::~MainWindow (this=0xbcfca0, __in_chrg=<optimized out>) at mainwindow.cpp:485
#7  0x000000000042332f in Zeal::Core::Application::~Application (this=0x7ffdc0, __in_chrg=<optimized out>) at application.cpp:96
#8  0x0000000000423396 in Zeal::Core::Application::~Application (this=0x7ffdc0, __in_chrg=<optimized out>) at application.cpp:98
#9  0x00000000004227cf in QScopedPointerDeleter<Zeal::Core::Application>::cleanup (pointer=0x7ffdc0) at /usr/include/x86_64-linux-gnu/qt5/QtCore/qscopedpointer.h:54
#10 0x0000000000422205 in QScopedPointer<Zeal::Core::Application, QScopedPointerDeleter<Zeal::Core::Application> >::~QScopedPointer (this=0x7fffffffdce0, __in_chrg=<optimized out>)
    at /usr/include/x86_64-linux-gnu/qt5/QtCore/qscopedpointer.h:101
#11 0x0000000000420a27 in main (argc=1, argv=0x7fffffffde58) at main.cpp:264
```

**Comment by trollixx** [comment]
I don't see anything specific :( Looks like `MainWindow`'s destruction failed, but no idea why...

If you know an older revision, when the problem didn't occur, `git bisect` would allow to locate the actual change, that caused the problem.

Thank you for spending time on debugging this, I really appreciate that.

**Comment by e-orz** [comment]
Sorry it took time. I used `git bisect` and the results are:
```
1c068d42f72ce16d7dea0f863ad903640d5a0492 is the first bad commit
commit 1c068d42f72ce16d7dea0f863ad903640d5a0492
Author: Oleg Shparber <oleg@zealdocs.org>
Date:   Sun Sep 11 02:14:48 2016 -0400

    qmake: Reorganize the whole project
    
    This is an attempt to provide a better structure to the project.
    All modules are now individual subprojects called libraries.
    QxtGlobalShortcut was moved to the Ui library, and resources are part
    of the application (app) subproject.

:100644 100644 5fbbff49db4e82b1b3a9e6632ca1557c91ddbcb3 e96e51b139957237e947305036cc4087dc75d414 M	.qmake.conf
:040000 040000 827b3d74d6ed1332c9b6ec1f70e68e9762404421 b4805d57dd27145f2aa02910fa6dc749f06c77c6 M	assets
:100644 000000 87e3486ad67881fb08460148c055a9b1f715fe35 0000000000000000000000000000000000000000 D	common.pri
:000000 040000 0000000000000000000000000000000000000000 194e0d2de9b094ea22a3fdaebb2375b5a09cc05a A	qmake
:040000 040000 9a6f8ad4b27e95b91eebcec45924dfcb1635125b 4e7480983d8c80acc092a01254c2b1870828daea M	src
```

Do you have any idea? This is a heavy commit.
I am happy to make the necessary tests or changes to find the root cause.

**Comment by trollixx** [comment]
Thanks!

This is the worst commit to cause a problem. Have you tried a completely clean build? If you build inside source tree, just run `git clean -xfd`, if outside, remove your build directory.

**Comment by trollixx** [comment]
And another thing to try would be commenting out lines in MainWindow's
destructor. I may be deleting something in a wrong order.

**Comment by trollixx** [comment]
I was able to reproduce the crash on my Ubuntu 14.04 VM, didn't have time to investigate the actual cause.

**Comment by e-orz** [comment]
Thanks!
I remember seeing somewhere that on Ubuntu 14.04 there is a known issue:
https://github.com/Swordfish90/cool-retro-term/issues/21- according to this site it's resolved on QT 5.3
I installed latest QT and used 5.7 - didn't help.

**Comment by trollixx** [comment]
So, it's something Ubuntu related, we just don't know what exactly yet. And according to your findings 1c068d42f72ce16d7dea0f863ad903640d5a0492 introduced that somehow...

**Comment by Farenjihn** [comment]
Hello,

This issue also appears for me on the following machine:

- OS: Archlinux
- Kernel: x86_64 Linux 4.7.5-1-ARCH

I can confirm that a clean build before commit https://github.com/zealdocs/zeal/commit/1c068d42f72ce16d7dea0f863ad903640d5a0492 clears all problems.

Building version number 0.3.0 manually or through the AUR yielded the same results (crash on close).

I quickly tried to check things out through GDB and got very similar results.

So it might not be related to Ubuntu specifically.

**Comment by trollixx** [comment]
@Farenjihn thanks for the information. I have exactly same setup, but no crash. Since we know the exact change, should be relatively easy to hunt the problem down, and release a fix.

**Comment by edouard-lopez** [comment]
Similar issue on `Zeal 0.3.0` pulled from the official PPA (`1:0.3.0-1ppa1~xenial1`)

```
$ lsb_release -a
No LSB modules are available.
Distributor ID: LinuxMint
Description:    Linux Mint 18 Sarah
Release:        18
Codename:       sarah
```

    Zeal 0.3.0

**Comment by peko** [comment]
Confirming, I have the same error.  Crashing after exit on fresh install.  After docset downloading - crashed right after start. 

OS: Arch Linux x86_64
Kernel Release: 4.7.5-1-ARCH

**Comment by stardiviner** [comment]
I still have this problem.
Environment:
```
Arch Linux: latest
Zeal version: 0.3.0 (latest version I built from git source code)
```
When I execute command in shell:
```
zeal --version
Zeal 0.3.0
[1]    31669 segmentation fault (core dumped)  zeal --version
````
You can see it cause `segmentation fault`. This output cause Emacs package `zeal-at-point` get Zeal version failed. Here is its source code: https://github.com/jinzhu/zeal-at-point/blob/master/zeal-at-point.el#L134

**Comment by trollixx** [comment]
@stardiviner Are you sure you built from the `hotfix/0.3.1` branch, and not `master`?

**Comment by stardiviner** [comment]
I tried the lates `hotfix/0.3.1` branch. and compile from it. Try again with command `zeal --version`. Still get output `segmentation fault`. But zeal will not crash now.

**Comment by trollixx** [comment]
@stardiviner Oh, I didn't realise it crashes even for `--version`. Then it's a different bug. Could you get a meaningful stack trace a create a new ticket?

**Comment by stardiviner** [comment]
```
(gdb) start
Temporary breakpoint 2 at 0x41ea14
Starting program: /home/stardiviner/Code/Emacs/zeal/bin/zeal
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/usr/lib/libthread_db.so.1".

Temporary breakpoint 2, 0x000000000041ea14 in main ()
(gdb) step
Single stepping until exit from function main,
which has no line number information.
operator new (sz=16) at /build/gcc/src/gcc/libstdc++-v3/libsupc++/new_op.cc:43
43      /build/gcc/src/gcc/libstdc++-v3/libsupc++/new_op.cc: No such file or directory.
(gdb)
```

**Comment by jkozera** [comment]
> I don't know how to gdb on zeal --version.

You need to pass `--version` to gdb's `run` commend. I.e., try something like:
```
zeal$ gdb ./bin/zeal 
(...)
(gdb) run --version
```
Now when you get the SIGSEGV, type:
```
(gdb) where
```
to get a traceback.

**Comment by stardiviner** [comment]
```
(gdb) start
Temporary breakpoint 1 at 0x41ea14
Starting program: /home/stardiviner/Code/Emacs/zeal/bin/zeal
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/usr/lib/libthread_db.so.1".

Temporary breakpoint 1, 0x000000000041ea14 in main ()
(gdb) run --version
The program being debugged has been started already.
Start it from the beginning? (y or n) y
Starting program: /home/stardiviner/Code/Emacs/zeal/bin/zeal --version
[Thread debugging using libthread_db enabled]
Using host libthread_db library "/usr/lib/libthread_db.so.1".
[New Thread 0x7fffdf559700 (LWP 4894)]
[New Thread 0x7fffd36a8700 (LWP 4895)]
Zeal 0.3.0

Thread 3 "QDBusConnection" received signal SIGSEGV, Segmentation fault.
[Switching to Thread 0x7fffd36a8700 (LWP 4895)]
0x00007ffff22c19cf in QObject::disconnect(QObject const*, char const*, QObject const*, char const*) () from /usr/lib/libQt5Core.so.5
(gdb) where
#0  0x00007ffff22c19cf in QObject::disconnect(QObject const*, char const*, QObject const*, char const*) () from /usr/lib/libQt5Core.so.5
#1  0x00007fffdfbde530 in ?? () from /usr/lib/libQt5DBus.so.5
#2  0x00007fffdfbcaf62 in ?? () from /usr/lib/libQt5DBus.so.5
#3  0x00007ffff20b7d78 in ?? () from /usr/lib/libQt5Core.so.5
#4  0x00007fffef247454 in start_thread () from /usr/lib/libpthread.so.0
#5  0x00007ffff17b67df in clone () from /usr/lib/libc.so.6
(gdb)
```

**Comment by stardiviner** [comment]
This maybe caused by KDE's `kded5` from here: https://bugs.kde.org/show_bug.cgi?id=370249

**Comment by jkozera** [comment]
Looks like https://bugreports.qt.io/browse/QTBUG-52988

**Comment by e-orz** [comment]
For me the hotfix/0.3.1 solved the issue and --version also works as expected.
Thanks!

**Comment by edouard-lopez** [comment]
@trollixx could you publish the `0.3.1` on launchpad/PPA?

**Comment by thuandt** [comment]
I'm waiting for `0.3.1` in PPA too.
Maybe he's busy, i guess we will have it on PPA at weekend

**Comment by trollixx** [comment]
PPA has been updated to v0.3.1.
